src/NServiceBus.Core/Scheduling/Schedule.cs

```diff
@@ -3,57 +3,83 @@ namespace NServiceBus
         /// <summary>
         /// Schedules a task to be executed repeatedly in a given interval.
         /// </summary>
-        /// <param name="bus">Bus.</param>
+        /// <param name="context">Bus.</param>
         /// <param name="timeSpan">The interval to repeatedly execute the <paramref name="task"/>.</param>
         /// <param name="task">The <see cref="System.Action"/> to execute.</param>
-        public static void ScheduleEvery(this IBusContext bus, TimeSpan timeSpan, Action task)
+        [ObsoleteEx(ReplacementTypeOrMember = "ScheduleEvery(this IBusContext context, TimeSpan timeSpan, Func<IBusContext, Task> task)")]
+        public static void ScheduleEvery(this IBusContext context, TimeSpan timeSpan, Action task)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Schedules a task to be executed repeatedly in a given interval.
+        /// </summary>
+        /// <param name="context">Bus.</param>
+        /// <param name="timeSpan">The interval to repeatedly execute the <paramref name="task"/>.</param>
+        /// <param name="task">The <see cref="System.Action"/> to execute.</param>
+        /// <param name="name">The name to use used for logging inside the new <see cref="Thread"/>.</param>
+        [ObsoleteEx(ReplacementTypeOrMember = "ScheduleEvery(this IBusContext context, TimeSpan timeSpan, string name, Func<IBusContext, Task> task)")]
+        public static void ScheduleEvery(this IBusContext context, TimeSpan timeSpan, string name, Action task)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Schedules a task to be executed repeatedly in a given interval.
+        /// </summary>
+        /// <param name="context">Bus.</param>
+        /// <param name="timeSpan">The interval to repeatedly execute the <paramref name="task"/>.</param>
+        /// <param name="task">The async function to execute.</param>
+        public static Task ScheduleEvery(this IBusContext context, TimeSpan timeSpan, Func<IBusContext, Task> task)
         {
             Guard.AgainstNull(nameof(task), task);
             Guard.AgainstNegativeAndZero(nameof(timeSpan), timeSpan);
-            var declaringType = task.Method.DeclaringType;
 
+            var declaringType = task.Method.DeclaringType;
             while (declaringType.DeclaringType != null &&
                 declaringType.CustomAttributes.Any(a => a.AttributeType.Name == "CompilerGeneratedAttribute"))
             {
                 declaringType = declaringType.DeclaringType;
             }
 
-            ScheduleEvery(bus, timeSpan, declaringType.Name, task);
+            return ScheduleEvery(context, timeSpan, declaringType.Name, task);
         }
 
         /// <summary>
         /// Schedules a task to be executed repeatedly in a given interval.
         /// </summary>
-        /// <param name="bus">Bus.</param>
+        /// <param name="context">Bus.</param>
```

Comment:
Perhaps flesh out the api doc? "Context that allows you to perform bus operations" or similar?